src/status_im2/common/resources.cljs

```diff
@@ -0,0 +1,46 @@
+(ns status-im2.common.resources)
+
+(def ui
+  {:add-new-contact (js/require "../resources/images/ui2/add-contact.png")
+   :lifestyle       (js/require "../resources/images/ui2/lifestyle.png")
+   :music           (js/require "../resources/images/ui2/music.png")
+   :podcasts        (js/require "../resources/images/ui2/podcasts.png")})
+
+(def mock-images
+  {:coinbase             (js/require "../resources/images/mock2/coinbase.png")
+   :collectible          (js/require "../resources/images/mock2/collectible.png")
+   :community-banner     (js/require "../resources/images/mock2/community-banner.png")
+   :community-logo       (js/require "../resources/images/mock2/community-logo.png")
+   :community-cover      (js/require "../resources/images/mock2/community-cover.png")
+   :gif                  (js/require "../resources/images/mock2/gif.png")
+   :photo1               (js/require "../resources/images/mock2/photo1.png")
+   :photo2               (js/require "../resources/images/mock2/photo2.png")
+   :photo3               (js/require "../resources/images/mock2/photo3.png")
+   :small-opt-card-icon  (js/require "../resources/images/mock2/small_opt_card_icon.png")
+   :small-opt-card-main  (js/require "../resources/images/mock2/small_opt_card_main.png")
+   :status-logo          (js/require "../resources/images/mock2/status-logo.png")
+   :sticker              (js/require "../resources/images/mock2/sticker.png")
+   :user-picture-female2 (js/require "../resources/images/mock2/user_picture_female2.png")
+   :user-picture-male4   (js/require "../resources/images/mock2/user_picture_male4.png")
+   :user-picture-male5   (js/require "../resources/images/mock2/user_picture_male5.png")})
+
+(def loaded-images (atom {}))
+
+(defn get-image
+  [k]
+  (if (contains? @loaded-images k)
+    (get @loaded-images k)
+    (get (swap! loaded-images assoc
+           k
+           (get ui k))
+         k)))
+
+(defn get-mock-image
+  [k]
+  (if (contains? @loaded-images k)
```

Comment:
I know this code was just copied over, but, I can't figure out what and why the functions `get-image` and `get-mock-image` are trying to cache/memoize... The vars `ui` and `mock-images` are already static maps, all `js/require` calls were already made and stored in memory. Caching the access to a static in-memory map is pointless.

Or am I missing something here?

The implementation can just be:

```clojure
(defn get-mock-image
  [k]
  (get mock-images k))

(defn get-image
  [k]
  (get ui k))

;; Also discard the `loaded-images` atom.
```

Reply:
yeah I'm looking at it too and wondering this. I guess we can always remove and then if we notice issues we can reconsider this code.
Or perhaps @flexsurfer or  @rasom has more insight of this code?

Reply:
Cool @J-Son89. I tested the suggested implementation on the REPL. It does the same as the original one and should have no impact in performance. It even performs better because there's no atom overhead ;)

Maybe in the past the memoization was needed when image vars weren't statically defined. Who knows, but now it's just overhead.

Reply:
don't think much about the past :) better to think about bright feature  :)

Reply:
Exactly @flexsurfer, that's what counts :heart:

Reply:
Nice catch @ilmotta! The less atoms the better, IMO.

@flexsurfer I love the sentiment! I think the recent [discussion](https://github.com/status-im/status-mobile/discussions/15099) you had with @ulisesmac is a good example of a practice that I can see being useful in this respect. You captured your reasoning (with a nice TDLR), which can:
- be linked to if the topic comes up again
- be revisited if/when the code/design is changed